rs/index/src/posting_list/mod.rs

```diff
@@ -26,12 +27,23 @@ pub struct PostingListIterator<'a> {
 }
 
 impl<'a> PostingList<'a> {
-    pub fn new_with_slices(slices: Vec<&'a [u8]>) -> Self {
-        PostingList { slices }
+    pub fn new_with_slices(slices: Vec<&'a [u8]>) -> Result<Self> {
+        let mut elem_count = 0;
+        let elem_size_in_bytes = size_of::<u64>();
+        for slice in &slices {
+            if slice.len() % elem_size_in_bytes != 0 {
+                return Err(anyhow!(
+                    "Invalid slice length {}: should be multiple of u64's size in bytes",
+                    slice.len()
+                ));
+            }
+            elem_count += slice.len() / elem_size_in_bytes;
+        }
+        Ok(PostingList { slices, elem_count })
     }
```

Comment:
my advice is to write the `count` into the disk format, and just read it. If you do this way, it works for now, but once you have Elias-Fano encoding, it won't work anymore

Reply:
yeah we are already writing that info to the disk format (in posting list metadata section, which consists of the posting list length (number of elements) and posting list starting offset)

This is to allow for quick debugging when building the index (i.e. working on the `PostingList` struct, which is a vector of (potentially disjoint) slices)